test/_init_tests.js

```diff
@@ -207,10 +207,31 @@ describe.configure = function() {
   return new TestConfig(describe);
 };
 
+const realIt = it;
+it = function(message, runnable) { // eslint-disable-line no-native-reassign, no-undef
+  if (!runnable || !runnable.constructor
+      || runnable.constructor.name !== 'GeneratorFunction') {
+    return realIt(message, runnable);
+  }
+  return realIt(message, done => {
+    const iterator = runnable();
+    const interval = setInterval(() => {
```

Comment:
@zhouyx This is the thread @lannka and I have been going back and forth over. The only proper fix to this is to yield the promise (this ensure its then blocks have been called).